lib/reverse_sentence.rb

```diff
@@ -1,6 +1,42 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
+# Time complexity: o(n)
+# Space complexity: o(1)
+def reverse_sentence(my_sentence) 
```

Comment:
Since you're building a reversed sentence string throughout this the space complexity is O(n), further there is an alternative solution which is also O(n) time complexity and O(1) space complexity.

Consider: Reverse the string completely. Then loop through the string reversing each individual word, in place (i.e. not making an array or another string). Think about how you might be able to do that.